cli/src/cli.rs

```diff
@@ -949,6 +953,37 @@ pub fn parse_command(
                 signers: vec![],
             })
         }
+        ("completion", Some(matches)) => {
+            match matches.value_of("shell") {
+                Some("bash") =>
+                    get_clap_app(
+                        crate_name!(), crate_description!(),
+                        solana_version::version!()
+                    ).gen_completions_to("solana", Shell::Bash, &mut stdout()),
+                Some("fish") =>
+                    get_clap_app(
+                        crate_name!(), crate_description!(),
+                        solana_version::version!()
+                    ).gen_completions_to("solana", Shell::Fish, &mut stdout()),
+                Some("zsh") =>
+                    get_clap_app(
+                        crate_name!(), crate_description!(),
+                        solana_version::version!()
+                    ).gen_completions_to("solana", Shell::Zsh, &mut stdout()),
+                Some("powershell") =>
+                    get_clap_app(
+                        crate_name!(), crate_description!(),
+                        solana_version::version!()
+                    ).gen_completions_to("solana", Shell::PowerShell, &mut stdout()),
+                Some("elvish") =>
+                    get_clap_app(
+                        crate_name!(), crate_description!(),
+                        solana_version::version!()
+                    ).gen_completions_to("solana", Shell::Elvish, &mut stdout()),
+                _ => unreachable!() // This is safe, since we assign default_value
+            }
```

Comment:
nit: could dereplicode this block by having the `match` only select the `Shell` variant, then make a single call to `gen_completions_to`